test/test_conditional_payment_processor.js

```diff
@@ -1,51 +1,50 @@
+const { toHex, padLeft, keccak256, asciiToHex, hexToAscii, hexToNumber, toBN, toWei, fromWei } = NewWeb3.utils
 const ConditionalPaymentProcessor = artifacts.require('ConditionalPaymentProcessor')
 const WETH9 = artifacts.require('WETH9')
-const contracts = [ConditionalPaymentProcessor]
 
 contract('ConditionalPaymentProcessor', function (accounts) {
     let conditionalPaymentProcessorFactory
     let etherToken
     let oracle, questionId, payoutSlotCount, conditionalPaymentProcessor
     let conditionId
 
-    before(testGas.createGasStatCollectorBeforeHook(contracts))
-    after(testGas.createGasStatCollectorAfterHook(contracts))
-
     before(async () => {
         conditionalPaymentProcessor = await ConditionalPaymentProcessor.deployed()
         etherToken = await WETH9.deployed()
 
         // prepare condition
         oracle = accounts[1]
+
         questionId = '0xcafebabecafebabecafebabecafebabecafebabecafebabecafebabecafebabe'
         payoutSlotCount = 2
         await conditionalPaymentProcessor.prepareCondition(oracle, questionId, payoutSlotCount)
 
         conditionId = keccak256(oracle + [questionId, payoutSlotCount].map(v => padLeft(toHex(v), 64).slice(2)).join(''))
-    })
-
-    it('should target the right conditional payment processor');
-    it('should be creatable when nonexistent');
+    });
 
     it('should have obtainable conditionIds if in possession of oracle, questionId, and payoutSlotCount', async () => {
-        assert.equal((await conditionalPaymentProcessor.getPayoutSlotCount(conditionId)).valueOf(), payoutSlotCount)
+        assert.equal((await conditionalPaymentProcessor.getPayoutSlotCount(conditionId)).valueOf(), payoutSlotCount);
+        assert.equal((await conditionalPaymentProcessor.payoutDenominator(conditionId)).valueOf(), 0);
     });
 
+    it('should not be able to prepare the same condition more than once', async () => {
+        await utils.assertRejects(conditionalPaymentProcessor.prepareCondition(oracle, questionId, payoutSlotCount), 'Transaction should have reverted.')
+    })
+
     it('should split and merge positions on payout slots', async () => {
-        // Mint all outcomes
         const buyer = 0
         const collateralTokenCount = 1e19
         await etherToken.deposit({ value: collateralTokenCount, from: accounts[buyer] })
         assert.equal(await etherToken.balanceOf.call(accounts[buyer]), collateralTokenCount)
 
         await etherToken.approve(conditionalPaymentProcessor.address, collateralTokenCount, { from: accounts[buyer] })
-        for(let i = 0; i < 10; i++)
-            await conditionalPaymentProcessor.splitPosition(etherToken.address, 0, conditionId, [0b01, 0b10], collateralTokenCount / 10, { from: accounts[buyer] })
+        
+        for(let i = 0; i < 10; i++) {
+            await conditionalPaymentProcessor.splitPosition(etherToken.address, asciiToHex(0), conditionId, [0b01, 0b10], collateralTokenCount / 10, { from: accounts[buyer] })
```

Comment:
So when splitting from collateral, you've changed all the zeroes to asciiToHex(0). However, the [docs](https://web3js.readthedocs.io/en/1.0/web3-utils.html#asciitohex) for web3 state that asciiToHex

> Returns the HEX representation of a given ASCII string.

The docs do not specify what happens when a Number is passed into asciiToHex. We should not rely on undocumented behavior of a function. I thought that a plain 0 should work in these cases. If not, then [toHex](https://web3js.readthedocs.io/en/1.0/web3-utils.html#tohex) or [numberToHex](https://web3js.readthedocs.io/en/1.0/web3-utils.html#numbertohex) should be more appropriate here.

Reply:
No, it seems to need exactly `asciiToHex(0)`, when I try to use `toHex(0)` for whatever reason it throws these errors:
`Error: Given parameter bytes has an invalid length: "0x0"`

This happened upon upgrading to Truffle 5.0, until then just plain 0 were fine.

Reply:
Haha okay yeah that's pretty weird. Anyway, asciiToHex gives you '0x00' when you give it the Number 0, so I guess bytes want an even number of hex digits or something.